src/lib/components/market/response_render_results_status.ts

```diff
@@ -0,0 +1,155 @@
+function overrideOnResponseRenderResults(proto: any) {
+    const originalMethod = proto.OnResponseRenderResults;
+
+    proto.OnResponseRenderResults = function(responseClass: any) {
+        const success = (responseClass && responseClass.responseJSON && responseClass.responseJSON.success) || false;
+        if (!success) {
+            paginationBox.showError();
+        } else {
+            paginationBox.hide();
+        }
+
+        originalMethod.call(this, responseClass);
+    };
+}
+
+function overrideGoToPage(proto: any) {
+    const originalMethodGoToPage = proto.GoToPage;
+
+    proto.GoToPage = function(page: number) {
+        paginationBox.showPending();
+        originalMethodGoToPage.call(this, page);
+    };
+}
+
+export function initResponseRenderResultsStatus() {
+    const id = setInterval(() => {
+        const searchResultsObject = g_oSearchResults;
+        if (!searchResultsObject) return;
+
+        const proto = Object.getPrototypeOf(searchResultsObject);
+
+        overrideOnResponseRenderResults(proto);
+        overrideGoToPage(proto);
+
+        clearInterval(id);
+    }, 1000);
+}
+
+class PaginationFailedBox extends HTMLElement {
```

Comment:
`PaginationFailedBox` seems to be a bit of a misleading term since it also showing "pending" status.

How about `PaginationStatusBox`?